File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
from itertools import product
def solution(word):
    # 다 만들고 정렬
    vowels = ['A', 'E', 'I', 'O', 'U']
    word_list = []
    for i in range(1, 6):
        for pr in product(vowels, repeat=i):
            word_list.append(''.join(pr))

    word_list.sort()
    
    return word_list.index(word) + 1

    
    
    

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string word) {
    int weights[5] = {781, 156, 31, 6, 1};
    string vowels = "AEIOU";
    
    int ans = 0;
    for (int i = 0; i < word.size(); i++) {
        for (int j = 0; j < 5; j++) {
            if(word[i] == vowels[j]) {
                ans += j * weights[i] + 1;
                break;
            }
        }
    }
    
    return ans;
}